provider/pkg/gen/overlays.go

```diff
@@ -1146,6 +1146,91 @@ var yamlConfigGroupResource = pschema.ResourceSpec{
 	},
 }
 
+//go:embed examples/overlays/configGroupV2.md
+var configGroupV2MD string
+
+var yamlConfigGroupV2Resource = pschema.ResourceSpec{
+	IsComponent: true,
+	ObjectTypeSpec: pschema.ObjectTypeSpec{
+		IsOverlay:   false,
+		Description: configGroupV2MD,
+		Properties: map[string]pschema.PropertySpec{
+			"resources": {
+				TypeSpec: pschema.TypeSpec{
+					Type: "array",
+					Items: &pschema.TypeSpec{
+						Ref: "pulumi.json#/Any",
+					},
+				},
+				Description: "Resources created by the ConfigGroup.",
+			},
+		},
+		Type: "object",
+	},
+	InputProperties: map[string]pschema.PropertySpec{
+		"files": {
+			TypeSpec: pschema.TypeSpec{
+				OneOf: []pschema.TypeSpec{
+					{
+						Type: "string",
+					},
+					{
+						Type: "array",
+						Items: &pschema.TypeSpec{
+							Type: "string",
+						},
+					},
+				},
+			},
+			Description: "Set of paths or a URLs that uniquely identify files.",
+		},
+		"objs": {
+			TypeSpec: pschema.TypeSpec{
+				OneOf: []pschema.TypeSpec{
+					{
+						Ref: "pulumi.json#/Any",
+					},
+					{
+						Type: "array",
+						Items: &pschema.TypeSpec{
+							Ref: "pulumi.json#/Any",
+						},
+					},
+				},
+			},
+			Description: "Objects representing Kubernetes resources.",
+		},
+		"resourcePrefix": {
+			TypeSpec: pschema.TypeSpec{
+				Type: "string",
+			},
+			Description: "An optional prefix for the auto-generated resource names. Example: A resource created with resourcePrefix=\"foo\" would produce a resource named \"foo-resourceName\".",
+		},
+		"skipAwait": {
+			TypeSpec: pschema.TypeSpec{
+				Type: "boolean",
+			},
+			Description: "Indicates that child resources should skip the await logic.",
+		},
+		"yaml": {
+			TypeSpec: pschema.TypeSpec{
+				OneOf: []pschema.TypeSpec{
+					{
+						Type: "string",
+					},
+					{
+						Type: "array",
+						Items: &pschema.TypeSpec{
+							Type: "string",
+						},
+					},
+				},
+			},
+			Description: "YAML text containing Kubernetes resource definitions.",
```

Comment:
Should we take this opportunity to rephrase this description? `Kubernetes resource definitions` is somewhat confusing to me as a native k8s user/developer. `Kubernetes manifests` makes more sense to me.

Reply:
It's true but, looking briefly at the k8 docs, seems like "resources" is used by kubectl documentation:
https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#create

Whereas other sections do use "manifest" as seen in the glossary:
https://kubernetes.io/docs/reference/glossary/?all=true#term-manifest
> Manifest - Specification of a Kubernetes API object in JSON or YAML format.[-]
A manifest specifies the desired state of an object that Kubernetes will maintain when you apply the manifest. Each configuration file can contain multiple manifests.